### PR TITLE
Add test for erroneous energy generating cycles

### PR DESCRIPTION
**Pull request by stelmo**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -13,6 +13,7 @@ JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
 MD5 = "6ac74813-4b46-53a4-afec-0b5dc9d7885c"
 Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+Tulip = "6dd1b50a-3aae-11e9-10b5-ef983d2400fa"
```
Comment (exaexa) [inline Project.toml:16]: I guess this isn't necessary here; package should _not_ depend on any precise optimizer.

**Review by exaexa — APPROVED** [review]
I guess all is okay. 2 notes, possible for future consideration

- could we add & remove all the reactions in 1 big shot, preferably not doing it in-place on a single kind of model? If someone passes in a huge model this is going to be cringe-level slow   (maybe cobrexa could have overlays for doing this easily... :D )
- regarding the exports, is `memote` going to be one exported function or many small ones? I'd vote for one that runs a test set, just as done originally in MetabolicModelTests.jl :headstone: 

Also please check the comments on `Project.toml`, these may deserve a fix

File: Project.toml (hunk 1 above)
Comment (stelmo) [reply Project.toml:16]: fixing

File: Project.toml [hunk 2]
```diff
@@ -23,6 +24,7 @@ MD5 = "0.2"
 [extras]
 Downloads = "f43a241f-c20a-4ad4-852c-f6b1247861c6"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+Tulip = "6dd1b50a-3aae-11e9-10b5-ef983d2400fa"
```
[the remark this answers is not in the document]
Comment (stelmo) [reply Project.toml:27]: fixing

**Comment by stelmo** [comment]
> I guess all is okay. 2 notes, possible for future consideration
> 
> * could we add & remove all the reactions in 1 big shot, preferably not doing it in-place on a single kind of model? If someone passes in a huge model this is going to be cringe-level slow   (maybe cobrexa could have overlays for doing this easily... :D )
> * regarding the exports, is `memote` going to be one exported function or many small ones? I'd vote for one that runs a test set, just as done originally in MetabolicModelTests.jl 🪦
> 
> Also please check the comments on `Project.toml`, these may deserve a fix

I am open to all of this. Right now I am just building the skeleton, later we can smoothen it :)

**Pull request merged**